src/components/PostDetails.tsx

```diff
@@ -1,107 +1,117 @@
-import React from 'react';
+import React, { useEffect, useState, useCallback, useMemo } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { deleteComment, getComments } from '../api/comments';
+import { Comment } from '../types/Comment';
 
-export const PostDetails: React.FC = () => {
-  return (
-    <div className="content" data-cy="PostDetails">
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
+type Props = {
+  selectedPost: Post | undefined;
+};
 
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
-        </div>
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isFormVisible, setIsFormVisible] = useState(false);
 
-        <div className="block">
-          <Loader />
+  useEffect(() => {
+    if (selectedPost) {
+      setIsFormVisible(false);
+      setIsLoading(true);
+      setIsError(false);
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+      getComments(selectedPost.id)
+        .then(setComments)
+        .catch(() => setIsError(true))
+        .finally(() => setIsLoading(false));
+    }
+  }, [selectedPost]);
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+  const addNewComment = useCallback((newComment: Comment) => {
+    setComments(prev => [...prev, newComment]);
+  }, []);
 
-          <p className="title is-4">Comments:</p>
+  const deleteCommentHandler = useCallback((commentId: number) => {
+    setComments(prev => prev.filter(comment => comment.id !== commentId));
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+    deleteComment(commentId).catch(() => {
+      // Опционально: можно добавить обработку ошибок для восстановления удалённого комментария
```

Comment:
Comment must be on English

Reply:
I`m use chatgpt because i want to better understand how writing nice code and logic. However i know than it bad practice. maybe)